src/MakerInterface.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+
+/**
+ * Interface that all maker commands must implement.
+ *
+ * @author Ryan Weaver <ryan@knpuniversity.com>
+ */
+interface MakerInterface
+{
+    /**
+     * Return the command name for your maker (e.g. make:report).
+     *
+     * @return string
+     */
+    public static function getCommandName(): string;
+
+    /**
+     * Configure the command: set description, input arguments, options, etc.
+     *
+     * By default, all arguments will be asked interactively. If you want
+     * to avoid that, use the $inputConfig->setArgumentAsNonInteractive() method.
+     *
+     * @param Command            $command
+     * @param InputConfiguration $inputConfig
+     */
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void;
+
+    /**
+     * Configure any library dependencies that your maker requires.
+     *
+     * @param DependencyBuilder $dependencies
+     */
+    public function configureDependencies(DependencyBuilder $dependencies): void;
+
+    /**
+     * If necessary, you can use this method to interactively ask the user for input.
+     *
+     * @param InputInterface $input
+     * @param ConsoleStyle   $io
+     * @param Command        $command
+     */
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void;
```

Comment:
I found only 1 maker needing to perform custom interactions to fill arguments. So should this really be part of the interface implemented by all makers ?